config/v1/types_infrastructure.go

```diff
@@ -1010,6 +1010,15 @@ type VSpherePlatformTopology struct {
 	// +kubebuilder:validation:Pattern=`^/.*?/vm/.*?`
 	// +optional
 	Folder string `json:"folder,omitempty"`
+
+	// template is the inventory path of the virtual machine or template
+	// that will be used for cloning.
+	// +openshift:enable:FeatureSets=CustomNoUpgrade;TechPreviewNoUpgrade
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^/.*?/vm/.*?`
+	// +optional
+	Template string `json:"template,omitempty"`
```

Comment:
Why does this need to be added to the failure domain, and what happens to those who have not got it set presently?

Reply:
It is needed because every failure domain needs it's own template as a single, global template may not be accessible in all failure domains.  Additionally, the template may be defined at install time in the install-config.yaml so we have to have a way of ingesting and associating a template with a given failure domain.